analyzers/src/SonarAnalyzer.CSharp/Rules/RedundantConditionalAroundAssignmentCodeFixProvider.cs

```diff
@@ -46,11 +47,24 @@ protected override Task RegisterCodeFixesAsync(SyntaxNode root, CodeFixContext c
             var condition = root.FindNode(diagnosticSpan) as ExpressionSyntax;
             var ifStatement = condition?.FirstAncestorOrSelf<IfStatementSyntax>();
 
-            if (ifStatement == null)
+            if (ifStatement != null)
+            {
+                return HandleIfStatement(root, context, ifStatement);
+            }
+
+            var switchExpression = root.FindNode(diagnosticSpan).FirstAncestorOrSelf((SyntaxNode x) => x.IsKind(SyntaxKindEx.SwitchExpression));
```

Comment:
This is a bit more readable for me, but feel free to leave it as is if you don`t agree with it.
```suggestion
            var switchExpression = root.FindNode(diagnosticSpan).FirstAncestorOrSelf<SyntaxNode>(x => x.IsKind(SyntaxKindEx.SwitchExpression));
```